src/ir_Coolix.cpp

```diff
@@ -98,112 +98,172 @@ void IRsend::sendCOOLIX(uint64_t data, uint16_t nbits, uint16_t repeat) {
 //   https://github.com/crankyoldgit/IRremoteESP8266/issues/484
 IRCoolixAC::IRCoolixAC(const uint16_t pin, const bool inverted,
                        const bool use_modulation)
-    : _irsend(pin, inverted, use_modulation) { stateReset(); }
-
-void IRCoolixAC::stateReset() { setRaw(kCoolixDefaultState); }
```

Comment:
`stateReset()` needs to be restored. It is a standard function that all of the IR A/C classes have, and it is used in the Unit Tests. i.e. The lack of this method is causing a compilation error.

Reply:
Hi David,

I've reverted back the stateReset.
I'll try to explain the motivation. 
Basically there are two types of commands to be sent, the special ones (Turbo, Led, Swing, Sleep, Off) and the basic (change Temperature, Mode, Fan).
The special commands (state) are transient and won't be persisted at normal_state attribute, meaning, just after execute that it isn't possible to check using get if they are executed or not.
The regular commands (state) are either transient on the current implementation and relies on recoverSavedState and updateSavedState, which can cause issues while sending command like on setMode, where recoverSavedState(); (line 298) is going to cleanup what was done previously by setFan() (for instance).
My aim is to separete between Transient (special) and Permanent (basic) state, using the remote attribute to represent just the Permanent and the other attributes to save what was changed with the Transient.
Regards,
Daniel Ribeiro

Reply:
> My aim is to separete between Transient (special) and Permanent (basic) state, using the remote attribute to represent just the Permanent and the other attributes to save what was changed with the Transient.

Gotcha. Thanks for the explanation.

> recoverSavedState and updateSavedState, which can cause issues while sending command like on setMode, where recoverSavedState(); (line 298) is going to cleanup what was done previously by setFan() (for instance).

Hmm. That sounds like a bug. `setMode()` & `setFan()` should be able to be called in any order and maintain the correct internal state. We can either fix it, or ... as you've headed towards, store the temp/mode/etc state in variables, and construct the `remote` state when `send()` or `getRaw()` is called. This is similar to the approach in `ir_Fujitsu.cpp`.

I do see more merit in the latter approach (i.e. your current approach), but it can lead to an awful mess. e.g. Fujitsu. :)

Reply:
Thank you for the reviews.
Got the point... will think in a less intrusive way to perform the improvement :)